File: CStoreManagement/app/src/main/java/com/c_store/zhiyazhang/cstoremanagement/presenter/signin/SignInPresenter.java
package com.c_store.zhiyazhang.cstoremanagement.presenter.signin;

import android.os.Handler;

import com.c_store.zhiyazhang.cstoremanagement.bean.UserBean;
import com.c_store.zhiyazhang.cstoremanagement.model.MyListener;
import com.c_store.zhiyazhang.cstoremanagement.model.signin.SignInInterface;
import com.c_store.zhiyazhang.cstoremanagement.model.signin.SignInModel;
import com.c_store.zhiyazhang.cstoremanagement.utils.ConnectionDetector;
import com.c_store.zhiyazhang.cstoremanagement.view.interfaceview.SignInView;

/**
 * Created by zhiya.zhang
 * on 2017/5/8 15:02.
 */

public class SignInPresenter {
    private SignInInterface signInInterface;
    private SignInView signInView;
    private Handler mHandler = new Handler();
    private ConnectionDetector cd = ConnectionDetector.getConnectionDetector();

    public SignInPresenter(SignInView signInView) {
        this.signInView = signInView;
        this.signInInterface = new SignInModel();
    }

    public void login() {
        if (!cd.isOnline()) {
            signInView.hideLoading();
            return;
        }
        signInView.showLoading();
        signInInterface.login(signInView.getUID(), signInView.getPassword(), new MyListener() {
            @Override
            public void contractSuccess() {

            }

            @Override
            public void contractSuccess(final Object object) {
                mHandler.post(new Runnable() {
                    @Override
                    public void run() {
                        signInView.toActivity((UserBean) object);
                        signInView.saveUser((UserBean) object);
                        signInView.hideLoading();
                    }
                });
            }

            @Override
            public void contractFailed(final String errorMessage) {
                mHandler.post(new Runnable() {
                    @Override
                    public void run() {
                        signInView.showFailedError(errorMessage);
                        signInView.hideLoading();
                    }
                });
            }
        });
    }
}

File: KotlinCStoreManagement/app/src/main/java/com/cstore/zhiyazhang/cstoremanagement/presenter/signin/SignInPresenter.kt
package com.cstore.zhiyazhang.cstoremanagement.presenter.signin

import android.os.Handler
import com.cstore.zhiyazhang.cstoremanagement.bean.User
import com.cstore.zhiyazhang.cstoremanagement.model.MyListener
import com.cstore.zhiyazhang.cstoremanagement.model.signin.SignInInterface
import com.cstore.zhiyazhang.cstoremanagement.model.signin.SignInModel
import com.cstore.zhiyazhang.cstoremanagement.utils.ConnectionDetector
import com.cstore.zhiyazhang.cstoremanagement.view.interfaceview.GenericView
import com.cstore.zhiyazhang.cstoremanagement.view.interfaceview.SignInView

/**
 * Created by zhiya.zhang
 * on 2017/5/8 15:02.
 */

class SignInPresenter(private val signInView: SignInView, val genericView: GenericView) {
    private val signInInterface: SignInInterface
    private val mHandler = Handler()

    init {
        this.signInInterface = SignInModel()
    }

    fun login() {
        if (!ConnectionDetector.getConnectionDetector().isOnline) {
            genericView.hideLoading()
            return
        }
        genericView.showLoading()
        signInInterface.login(signInView.uid, signInView.password, object : MyListener {
            override fun listenerSuccess() {

            }

            override fun listenerSuccess(`object`: Any) {
                mHandler.post {
                    genericView.requestSuccess(`object` as User)
                    signInView.saveUser(`object`)
                    genericView.hideLoading()
                }
            }

            override fun listenerFailed(errorMessage: String) {
                mHandler.post {
                    genericView.showPrompt(errorMessage)
                    genericView.hideLoading()
                }
            }
        })
    }
}
